functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -233,6 +233,20 @@ struct LinalgCheckMatrixBinaryRuleHelper;
 
 template <char const *op_name, typename F, F Func, typename A, typename B, typename... T>
 struct LinalgCheckMatrixBinaryRuleHelper<op_name, F, Func, typelist<A, B, T...>> {
+  static oneOutput apply_one(
+      const Tensor& first, optional<int64_t> first_bdim,
+      const Tensor& second, optional<int64_t> second_bdim,
+      T... extra_args) {
+    TORCH_CHECK(!first_bdim.has_value() || first.dim() > 2,
+              op_name, ": The input tensor A must have at least 2 dimensions.");
+    TORCH_CHECK(!second_bdim.has_value() || second.dim() > 2,
+              op_name, ": The input tensor A must have at least 2 dimensions.");
+    const auto tensor_other = _binary_pointwise_helper(first, first_bdim, second, second_bdim);
```

Comment:
Same comments as the last PR: should compute rank without bdim and use that, and also binary_pointwise_helper does type promotion